contributor-docs/testing.md

```diff
@@ -5,19 +5,88 @@
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 ## Table of Contents
 
-- [Overview](#overview)
-- [Prerequisites](#prerequisites)
-- [Components](#components)
-  - [Visual Regression Testing](#visual-regression-testing)
-  - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Strategy](#strategy)
+- [Unit Tests](#unit-tests)
+  - [Overview](#overview)
+  - [What We Test](#what-we-test)
+  - [Running Tests](#running-tests)
+- [Visual Regression Tests](#visual-regression-tests)
+  - [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Components](#components)
+    - [Visual Regression Testing](#visual-regression-testing)
+    - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Interaction Tests](#interaction-tests)
+  - [As A Part Of Unit Tests](#as-a-part-of-unit-tests)
+  - [Storybook Interaction Tests](#storybook-interaction-tests)
+- [Accessibility Tests](#accessibility-tests)
 - [Continous Integration](#continous-integration)
 - [FAQ](#faq)
   - [Why am I seeing `browserType.launch: Executable doesn't exist at ../path`?](#why-am-i-seeing-browsertypelaunch-executable-doesnt-exist-at-path)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 <!-- prettier-ignore-end -->
 
-## Overview
+##Strategy
+
+Strategy
```

Comment:
I need some help here. Especially from folks who has been in the team for a while. How would you describe our strategy to testing?

Reply:
Here are some bullet points based on my experience on the team:
- We cover all use-cases within reason. We strive for 100% coverage, but 
- Tests should catch regressions in functionality, visual design, and user experience in general
  - Regressions in functionality are caught by unit tests and interaction tests executed using [Storybook Play functions](https://storybook.js.org/docs/react/writing-stories/play-function)
  - Regressions in visual design are caught by Chromatic (and sometimes by snapshots)
  - Regressions in user experience are caught by a11y linting of snapshots and interaction tests executed using [Storybook Play functions](https://storybook.js.org/docs/react/writing-stories/play-function)
- Tests can help us avoid making the same mistake twice. When we find a bug that wasn't caught by tests, we write a test to cover that case. (I guess this is related to the last point about regressions)

Reply:
> Tests should catch regressions in functionality, visual design, and user experience in general 

I love it. This is very comprehensive yet simplified 💖

Reply:
@mperrotti nailed it. 

A few other things for historical context...

- Snapshot testing with React components in the past has led to artificially inflated test coverage on certain components. Unfortunately these won't provide the required resiliency against regressions, so maintainers agreed to either use them as supplementary test coverage to a traditional unit test or avoid it altogether. This was discussed in an eng sync last year, but we never reached the critical mass on removing it entirely. 

- I'd refer to the transition away from enzyme to behavioural testing paradigms. So less testing against internal mechanics and rather focussing on the input/output. The testing library docs are a good enough reference point, and we should also probably consider adopting most of their recommendations for matchers and assertions.

- Unit tests should cover happy _and_ unhappy paths, with localised mocks where necessary. While we don't follow TDD formally, it could help for designers and engineers to be aware of all unhappy paths before building out components so that those scenario's are captured in our test suites.

Reply:
Ah, thank you so much you all, love them! I'll collate them together and put something up for our strategy 🚀

Reply:
I wrote the strategy section 😅 I tried to cover all of your amazing points here and some of them like tool specific regression tests are mentioned under other sections. I would appreciate if you could give it a read and let me know what you think. Anything I missed, or misinterpreted, or got it completely wrong, or any grammar mistakes, typo all are appreciated 🙏🏼